ompi/mca/common/ompio/common_ompio_file_read.c

```diff
@@ -352,53 +348,103 @@ int mca_common_ompio_file_read_at (ompio_file_t *fh,
     return ret;
 }
 
+static void mca_common_ompio_post_next_read_subreq(struct mca_ompio_request_t *req, int index)
+{
+    uint32_t iov_count = 1;
+    size_t pos = 0, spc = 0, tbw = 0;
+    int i = 0;
+    mca_ompio_request_t *ompio_subreq=NULL;
+    size_t bytes_per_cycle = OMPIO_MCA_GET(req->req_fh, pipeline_buffer_size);
+
+    struct iovec *decoded_iov = (struct iovec *) malloc (sizeof (struct iovec));
```

Comment:
Can this struct be on the stack to avoid heap allocation? Looks like it's freed at the end of the function.